dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/programrule/implementers/enrollment/SetMandatoryFieldExecutorTest.java

```diff
@@ -0,0 +1,233 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.programrule.implementers.enrollment;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.hisp.dhis.DhisConvenienceTest;
+import org.hisp.dhis.dataelement.DataElement;
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.program.ProgramStageDataElement;
+import org.hisp.dhis.program.ValidationStrategy;
+import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
+import org.hisp.dhis.tracker.TrackerIdSchemeParam;
+import org.hisp.dhis.tracker.TrackerIdSchemeParams;
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Attribute;
+import org.hisp.dhis.tracker.domain.Enrollment;
+import org.hisp.dhis.tracker.domain.EnrollmentStatus;
+import org.hisp.dhis.tracker.domain.MetadataIdentifier;
+import org.hisp.dhis.tracker.preheat.TrackerPreheat;
+import org.hisp.dhis.tracker.programrule.IssueType;
+import org.hisp.dhis.tracker.programrule.ProgramRuleIssue;
+import org.hisp.dhis.tracker.validation.ValidationCode;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+@ExtendWith( MockitoExtension.class )
+class SetMandatoryFieldExecutorTest extends DhisConvenienceTest
+{
+    private final static String ACTIVE_ENROLLMENT_ID = "ActiveEnrollmentUid";
+
+    private final static String COMPLETED_ENROLLMENT_ID = "CompletedEnrollmentUid";
+
+    private final static String DATA_ELEMENT_ID = "DataElementId";
+
+    private final static String ATTRIBUTE_ID = "AttributeId";
+
+    private final static String ATTRIBUTE_CODE = "AttributeCode";
+
+    private final static String TEI_ID = "TeiId";
+
+    private final static String ATTRIBUTE_VALUE = "23.0";
+
+    private static ProgramStage firstProgramStage;
+
+    private static ProgramStage secondProgramStage;
+
+    private static DataElement dataElementA;
+
+    private static DataElement dataElementB;
+
+    private TrackedEntityAttribute attribute;
+
+    private final SetMandatoryFieldExecutor enrollmentValidatorToTest = new SetMandatoryFieldExecutor( "RULE_ATTRIBUTE",
```

Comment:
Our type names are generally pretty long and thus take up a lot of space. The prefix like `enrollment...` is known from the package already. I like to keep the vars as short as possible without loosing information.

I would call it `validator`. This makes lines like

```java

        Optional<ProgramRuleIssue> error = enrollmentValidatorToTest.executeEnrollmentRuleAction( bundle,
            getEnrollmentWithMandatoryAttributeSet() );
```

```java

        Optional<ProgramRuleIssue> error = validator.executeEnrollmentRuleAction( bundle,
            getEnrollmentWithMandatoryAttributeSet() );
```

a little less verbose without loosing information IMHO.